llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4017,6 +4015,16 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
       }
     }
 
+    // For SMIN, if CstHigh is negative we know the result will be
```

Comment:
I still don't understand why this part is needed. Does it affect codegen for any tests?

Reply:
Yes